build.gradle

```diff
@@ -211,6 +207,19 @@ String intellijSetupGroup = 'IntelliJ IDEA Setup'
 
 // IntelliJ Project Setup
 
+// Use a different buildDir for IntelliJ builds to avoid conflict with CLI builds
+if (System.getProperty('idea.executable') != null) {
```

Comment:
I remember while I was trying to do this previously I was trying to look for a property `IntelliJ` specifically set so we don't have to set the VM options but I ended up finding implementation details (which means they are subject to change unless it is documented). Can I know where you found this and if it is documented, this looks unlikely to change though.

Reply:
I didn't find it from documentation, but from printing out all system properties, so yes it's implementation specific 😅 I still think it's better than asking the user to set a VM option just for the script to detect IntelliJ though, even with the risk that it might change.

Reply:
Why is it better than asking the user to set the VM option though? Is there something particularly bad about setting the VM option since this VM specifically only runs Gradle from IntelliJ IDEA?

What I am afraid is that different IntellIJ versions have different properties, like older IntelliJs vs new IntelliJs. Even if we find out that there's not so much problem, I think we have to document this properly otherwise this kinda looks magic lol.

Reply:
@wkurniawan07 what is the reason you didn't like setting VM options though? While I would have preferred something else, VM options seemed like the best option.

Reply:
I feel it's too much hassle for the user.

Reply:
It seems like just one step but I'm personally for automating this step away haha.

But it seems like this is something a maintainer would have to take special care of and use some heuristics to determine if IntelliJ has changed that setting.

It could go from just waiting for a person to report that their IDE setup is not working to checking on every release cycle or at least major versions of IntelliJ. Or maybe we can just ask Jetbrains if they would/won't ever change this.

But because this is only one step, it seems like a lot of work for us to keep this working although we know that it is guaranteed to work now and probably for the foreseeable future, past versions of IntelliJ are likely irrelevant because many things won't work anyway.

I don't have strong opinions on this though, just that I personally won't do this.

Reply:
@LiHaoTan developer's inconvenience is better than user's inconvenience. Of course where to draw the line is debatable, and since I'm unlikely to be a user of this routine (thanks to Eclipse's now incredible support), I'll leave it to you to decide.

Reply:
@whipermr5 seems like I don't have to block this if you prefer this? But you can add a comment along the lines of
// idea.executable is a special system property that IntelliJ sets when Gradle is run from IntelliJ
// Developers have to take special note of this property that IntelliJ sets as it is not documented and may change anytime.

Reply:
I just realised `idea.executable` isn't present in Windows 😅 Considering between `idea.paths.selector` (present on both macOS and Windows but I haven't verified on Linux) and `jb.vmOptionsFile` (I believe it's present everywhere across all JetBrains IDES due to [this](https://github.com/JetBrains/intellij-community/blob/9c86885ea0ecd3149b9c27fc066d5ce942337ee1/platform/platform-impl/src/com/intellij/diagnostic/VMOptions.java#L170-L175) and [this](https://www.jetbrains.com/help/idea/tuning-intellij-idea.html#d819828e396); however, it not being specific to IntelliJ may also be a drawback and we're looking at IntelliJ detection here).

Reply:
I got inspired and tried to see if IntelliJ documented some of their system properties (Not all are set by default). And it seemed they did.

https://www.jetbrains.com/help/idea/file-idea-properties.html
https://intellij-support.jetbrains.com/hc/en-us/articles/206544869-Configuring-JVM-options-and-platform-properties

The drawback is that these most likely apply to all their IDEs but I wonder if `idea.paths.selector` apply to other IDEs too.

I do think we shouldn't worry about this because all their IDEs are based off a single codebase with some differences, trying too hard to detect which IDE it is does not have much value because I suspect our project can be imported into Webstorm for example but without Java support probably.

Do you want to use one of them instead?

Reply:
I did read those articles when researching; the problem is that there is no guarantee that those properties are set. This is because those articles are documenting how to change certain IDE behaviour by setting those properties. They are _read_, not set, by the IDE. I've already extracted the full list of system properties set by IntelliJ by adding a gradle task to print them out and running it from within IntelliJ on Windows, macOS and Linux. Out of them all, `idea.paths.selector` seems like the most appropriate.

Reply:
Yeah, I am aware it is set by IntelliJ, but I wanted something documented. As of right now though, it seems like `idea.paths.selector` is the best for now 👍